src/helper/remote_config/client.cpp

```diff
@@ -179,6 +188,8 @@ bool client::process_response(const protocol::get_configs_response &response)
         }
     }
 
+    for (auto &listener : listeners_) { listener->commit(); }
```

Comment:
I think there is a small problem here. Let say a listener partially fails. I checked `asm_data_agregator`. If there is an exception within `asm_data_aggregator::add`, it may be the case that `rules_data_` gets partially populated and on`aggregate` it partially applies the config. However, the config would have been reported to RC as `error` applying. You may want to have a look this scenario in all aggregators

Reply:
As discussed, atomic updates will be future work.